app/main.py

```diff
@@ -1,5 +1,86 @@
+from typing import Optional
+
+
 class Cargo:
-    def __init__(self, weight):
+    def __init__(
+            self,
+            weight: int
+    ) -> None:
+
         self.weight = weight
 
-# write your code here
+
+class BaseRobot:
+    def __init__(
+            self,
+            name: str,
+            weight: int,
+            coords: Optional[list] = None
+    ) -> None:
+
+        self.name = name
+        self.weight = weight
+        if coords is None:
+            self.coords = [0, 0]
+        else:
+            self.coords = coords
+
+    def go_forward(self, step: int = 1) -> None:
+        self.coords[1] += step
+
+    def go_back(self, step: int = 1) -> None:
+        self.coords[1] -= step
+
+    def go_right(self, step: int = 1) -> None:
+        self.coords[0] += step
+
+    def go_left(self, step: int = 1) -> None:
+        self.coords[0] -= step
+
+    def get_info(self) -> str:
+        return f"Robot: {self.name}, Weight: {self.weight}"
+
+
+class FlyingRobot(BaseRobot):
+    def __init__(
+            self,
+            name: str,
+            weight: int,
+            coords: Optional[list] = None
+    ) -> None:
+
+        if coords is None:
+            super().__init__(name, weight, [0, 0, 0])
+        else:
+            super().__init__(name, weight, coords)
```

Comment:
Один раз виклич __init__ без кооридинат, і після цього перевизнч
```suggestion
        super().__init__(name, weight)
        self.coords = coords or [0,0,0]
```